lib/ansible/utils/module_docs_fragments/opennebula.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 www.privaz.io Valletech AB
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+class ModuleDocFragment(object):
+    # OpenNebula common documentation
+    DOCUMENTATION = '''
+options:
+    endpoint:
```

Comment:
Thanks for the feedback. Yes, we really need to agree on these ones for all the modules.

And we should make it a part of the design principles at: https://github.com/privazio/ansible/wiki/OpenNebula-Ansible-Design-Guidelines

I deliberately chose session following the XMLRPC spec where:

```
All methods expect a session string associated to the connected user as the first parameter.
```

Be aware that Username/Password are specific to the "core" authentication backed. OpenNebula supports other authentication mechanisms, that is why the default nomenclature is a bit generic.

I expect that the token mechanism will be used more than username/password, as tokens can be issued, traced and revoked from the console. 

So I think we should go with SESSION.

However, It would be possible to still take the username / password parameters in order to make it more comfortable for those users that use this authentication backend, as session = "username:password".

The ENDPOINT nomenclature comes from one.conf, the configuration file were users will have to find out were the connection endpoint is, however I don't think URL has any problems either. We just have to pick one. 

For environment variables, I think we should prefix with ONE_* just like the modules are.

What do you think?

Reply:
Looking at the documentation the `session` can either be `username:password` or `username:token`, so I would recommend to make it easier on the users of the module and just have username/password separate. Which means have `api_username`/`api_password` (like other modules have already) and just add an alias to this parameter for `api_token` and an additional `ONE_TOKEN` variable. We could easily modify other modules to have those options, until we switch them to `pyone`.

Regarding endpoint, I'm fine with having an alias for that as well: `api_url` and alias `api_endpoint` (or just `endpoint`). The reason why we decided to use the `api_*` prefix for these connections parameters is because that way in the module docs they will be grouped together (and mostly on top) in the list of parameters, so they are easy to see and understand what they mean (even if you never used OpenNebula).

Reply:
Looking at the documentation the `session` can either be `username:password` or `username:token`, so I would recommend to make it easier on the users of the module and just have username/password separate. Which means have `api_username`/`api_password` (like other modules have already) and just add an alias to this parameter for `api_token` and an additional `ONE_TOKEN` variable. We could easily modify other modules to have those options, until we switch them to `pyone`.

Regarding endpoint, I'm fine with having an alias for that as well: `api_url` and alias `api_endpoint` (or just `endpoint`). The reason why we decided to use the `api_*` prefix for these connections parameters is because that way in the module docs they will be grouped together (and mostly on top) in the list of parameters, so they are easy to see and understand what they mean (even if you never used OpenNebula).

Reply:
( I think I already replied this, but now I dont see it)

To be honest I did not like api_*
but you do have a point. I think getting all parameters together and on the top actually brings some value.
We can use the aliases. 
And I am concerned about leaving out session (for other core modules), however, we can add it up later based on field experience.
I will pull an update when I get a chance.

Reply:
Awesome! While we were working on our modules we looked at many different other cloud modules and we just checked what were the best things those modules had and copied from them, but also tried to fix the things that were bad. I'm also not the biggest fan of `api_*` names of parameters, they don't feel that they are about OpenNebula, but there are two big upsides which are far more important than that feel:
* When you read them you instantly know for what they are used from their names.
* They are grouped together (and on top) so they are easy to see and it's important that they are easy to see because without them, nothing works.

That's why in the end we decided to call them like that. I think we changed the names of those params during development at least 3 times until we settles with these :)